dev/docker-compose.yaml

```diff
@@ -0,0 +1,75 @@
+version: '3'
+services:
+
+    opensearch:
+        image: opensearchproject/opensearch:latest
+        container_name: opensearch
+        environment:
+            - cluster.name=opensearch-cluster
+            - node.name=opensearch
+            - discovery.type=single-node
+            - DISABLE_SECURITY_PLUGIN=true
+            - bootstrap.memory_lock=true # along with the memlock settings below, disables swapping
+        ulimits:
+            memlock:
+                soft: -1
+                hard: -1
+            nofile:
+                soft: 65536 # maximum number of open files for the OpenSearch user, set to at least 65536 on modern systems
+                hard: 65536
+        volumes:
+            - opensearch-data1:/usr/share/opensearch/data
+            - ../logs:/usr/share/opensearch/logs
+        ports:
+            - '9200:9200'
+            - '9600:9600' # required for Performance Analyzer
+        networks:
+            - opensearch-net
+    api-dummy:
+        image: elifesciences/api-dummy:29d80e7726a513afb2362068de55a9bab7aad0ae
+        container_name: api-dummy
+        ports:
+            - '5002:8080'
+        networks:
+            - opensearch-net
+    gearman:
+        image: artefactual/gearmand:latest
+        container_name: gearman
+        ports:
+            - '4730:4730'
+        networks:
+            - opensearch-net
+    goaws:
+        image: pafortin/goaws
+        container_name: goaws
+        volumes:
+            - ./.docker/goaws.yaml:/conf/goaws.yaml
+            - ./.docker/.aws:/root/.aws
+        ports:
+            - '4100:4100'
+        networks:
+            - opensearch-net
+    app:
+        build:
+            context: .
+            dockerfile: Dockerfile
+        volumes:
+            - ..:/app
+            - ./config.php:/app/config.php
```

Comment:
I haven't confirmed that this works yet. I am on a train at the moment so I will try this out when I'm not on the move.

Reply:
This seems to be working as expected.

Reply:
What is the reason for putting `config.php.dist` into `dev` folder? If someone wants to run the app with `builder`, they need to copy it into the root folder. Anyhow, if we decide to keep it in `dev`, then, we should change this line to `- ./config.php.dist:/app/config.php`